docker/Dockerfile.amd

```diff
@@ -2,72 +2,129 @@
 # This Dockerfile contains AMD compilers
 ##############################################################
 
+ARG IMAGE="rocm/dev-ubuntu-22.04:5.4.2-complete"
 ARG arch="aomp"
 
-ARG ROCM_VERSION=5.4.2
-FROM rocm/dev-ubuntu-22.04:${ROCM_VERSION}-complete as sdk-base
+FROM ${IMAGE} as sdk-base
+
+ARG UCX_BRANCH="v1.13.1"
+ARG OMPI_BRANCH="v4.1.4"
+
+# Update and Install basic Linux development tools
+RUN rm /etc/apt/sources.list.d/* \
+  && apt-get update \
+  && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    ca-certificates \
+    git \
+    ssh \
+    make \
+    vim \
+    nano \
+    libtinfo* \
+    initramfs-tools \
+    libelf-dev \
+    numactl \
+    wget \
+    tmux \
+    build-essential \
+    autoconf \
+    automake \
+    libtool \
+    pkg-config \
+    libnuma* \
+    gfortran \
+    flex \
+    hwloc \
+  && apt-get clean \
+  && ln -s /opt/rocm/llvm/bin/offload-arch /opt/rocm/bin/offload-arch
+
+# Requires cmake > 3.22 
+RUN mkdir -p /opt/cmake  \
+  && wget --no-check-certificate --quiet -O - https://cmake.org/files/v3.22/cmake-3.22.2-linux-x86_64.tar.gz | tar --strip-components=1 -xz -C /opt/cmake
+
+ENV ROCM_HOME=/opt/rocm \
+    UCX_HOME=/opt/ucx \
+    OMPI_HOME=/opt/ompi
+
+# Adding rocm/cmake to the Environment 
+ENV PATH=$ROCM_HOME/bin:$ROCM_HOME/profiler/bin:$ROCM_HOME/opencl/bin:/opt/cmake/bin:$PATH \
+    LD_LIBRARY_PATH=$ROCM_HOME/lib:$ROCM_HOME/lib64:$ROCM_HOME/llvm/lib:$LD_LIBRARY_PATH \
+    LIBRARY_PATH=$ROCM_HOME/lib:$ROCM_HOME/lib64:$LIBRARY_PATH \
+    C_INCLUDE_PATH=$ROCM_HOME/include:$C_INCLUDE_PATH \
+    CPLUS_INCLUDE_PATH=$ROCM_HOME/include:$CPLUS_INCLUDE_PATH \
+    CPATH=$ROCM_HOME/include:$CPATH \
+    INCLUDE=$ROCM_HOME/include:$INCLUDE
+
+WORKDIR /tmp
+
+# Install UCX
+RUN cd /tmp/ \
+    && git clone https://github.com/openucx/ucx.git -b ${UCX_BRANCH} \
+    && cd ucx \
+    && ./autogen.sh \
+    && mkdir build \
+    && cd build \
+    && ../contrib/configure-release --prefix=$UCX_HOME \
+        --with-rocm=$ROCM_HOME \
+        --without-knem \
+        --without-xpmem  \
+        --without-cuda \
+        --enable-optimizations  \
+        --disable-logging \
+        --disable-debug \
+        --disable-examples \
+    && make -j $(nproc)  \
+    && make install
+
+# Install OpenMPI
+RUN cd /tmp \
+    && git clone --recursive https://github.com/open-mpi/ompi.git -b ${OMPI_BRANCH} \
+    && cd ompi \
+    && ./autogen.pl \
+    && mkdir build \
+    && cd build \
+    && ../configure --prefix=$OMPI_HOME --with-ucx=$UCX_HOME \
```

Comment:
If you don't build openmpi against amdclang you need to change the amdclang compiler to have `self.mpicc` as amdclang or it's gonna compile with gcc